ensime_shared/ensime.py

```diff
@@ -109,13 +111,13 @@ def create_client(self, config_path):
 
         This will launch the ENSIME server for the project as a side effect.
         """
-        server_v2 = self.using_server_v2()
-        editor = Editor(self._vim)
-        launcher = EnsimeLauncher(self._vim, config_path, server_v2)
-        if server_v2:
-            return EnsimeClientV2(editor, self._vim, launcher)
+        config = ProjectConfig(config_path)
+        launcher = EnsimeLauncher(self._vim, config)
+
+        if self.using_server_v2:
```

Comment:
Hmm, you have all valid points, getting the version from the `ConnectionInfo` sound perfect, even more, we need the protocol version no the server version, it's only that this time internal change to the server were the perfect excuse to also change the protocol, but that could not be the case in the future.

The problem I see is that handling websocket messages is currently the main responsibility of the client, so we have a chicken and egg problem here due to the current implementation of the client  relaying on inheritance to pick the protocol version. IMHO the client should be only one, having all the WS machinery and after shacking hands with the server it should instantiate the proper version of protocol handlers.

Go ahead and create the ticket, the change looks big enough to be out of scope of this PR.